python/nav/portadmin/handlers.py

```diff
@@ -278,6 +290,24 @@ def is_configurable(self) -> bool:
             return False
         return True
 
+    def get_poe_state_options(self) -> Tuple[PoeState, ...]:
+        """Returns the available options for enabling/disabling PoE on this netbox"""
+        raise NotImplementedError
+
+    def set_poe_state(self, interface: manage.Interface, state: PoeState):
+        """Set state for enabling/disabling PoE on this interface.
+        Available options should be retrieved using `get_poe_state_options`
+        """
+        raise NotImplementedError
+
+    def get_poe_state(self, interface: manage.Interface) -> PoeState:
+        """Returns current PoE state of this interface"""
+        raise NotImplementedError
+
+    def interface_supports_poe(self, interface: manage.Interface) -> bool:
+        """Returns True if this interface supports PoE"""
+        raise NotImplementedError
+
```

Comment:
When loading portadmin, youre gonna want to call get_poe_state for all  poe enabled ports. Doing it one by one is not very efficient. Perhaps adding another function for getting ALL poe states would make sense. If this function returns poe state for all interfaces that support poe, then this response can be used to find out which interfaces do not support poe, making the interface_supports_poe function largely irrelevant. 

Keeping the smaller functions for getting information about a specific interface would not hurt, but having functions that do the heavy lifting more efficient would definitely be useful

Reply:
IIRC, some methods already have signatures that allow for fetching interface information from a range of interfaces, like this:

https://github.com/Uninett/nav/blob/5d97c4012f0f95ae2ebf6ed8b0b2136b5cc41356/python/nav/portadmin/handlers.py#L59-L72

I think that could be a useful pattern to re-use.  In these methods, the sequence of interfaces is an optional argument.  If omitted, the method will operate on all known interfaces.

Reply:
This seems quite useful, I'll prob add something like this